src/engine/TPA.cc

```diff
@@ -1589,8 +1592,16 @@ VerificationResult TransitionSystemNetworkManager::solve() && {
     while (true) {
         if (getNode(current).blocked_children == getNode(current).children.size()) {
             if (current == graph.getEntry()) {
-                // TODO: Compute witness
-                return {VerificationAnswer::SAFE, NoWitness{"Unable to compute witness from TPA for DAG of TSs yet"}};
+                if (Options::COMPUTE_WITNESS == "true") {
+                    auto witness = computeValidityWitness();
+                    if (holds_alternative<ValidityWitness>(witness)) {
+                        return {VerificationAnswer::SAFE, get<ValidityWitness>(witness)};
+                    } else {
+                        return {VerificationAnswer::SAFE, NoWitness()};
```

Comment:
```suggestion
                        return {VerificationAnswer::SAFE, std::get<NoWitness>(witness)};
```

Otherwise you would be throwing away the reason why no witness was computed.